fix(DB/Creature): Update En'kilah Hatchling Faction

cTrinity opened the pull request:
update En'kilah Hatchling(id = 25388,25390) unit_flags add 256 (Unable to fight or communicate with players)

## Changes Proposed:
-  update En'kilah Hatchling unit_flags
-  

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15241

## SOURCE:


## Tests Performed:
En'kilah Hatchling cannot be selected, cannot be attacked, and will not attack players
- 
- 


## How to Test the Changes:
Update SQL

## Known Issues and TODO List:
- [ ]
- [ ]

<!-- If you intend to contribute repeatedly to our project, it is a good idea to join our discord channel. We set ranks for our contributors and give them access to special resources or knowledge: https://discord.com/invite/DasJqPba)
     Do not remove the instructions below about testing, they will help users to test your PR -->
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

heyitsbench commented:
As with the last PR, what is the source for this change?

heyitsbench commented:
For the record, npcflag for 25390 is 0 for build 46158. This PR is wrong.

heyitsbench commented:
I apologize.I was mistaken on the concerned column for the template, the sniffed unit_flags for 25390 on build 46158 is 33554432. But again, that's already what's in AC's database, and this PR is still incorrect.

cTrinity commented:
> I apologize. I was mistaken on the concerned column for the template, the sniffed unit_flags for 25390 on build 46158 is 33554432. But again, that's already what's in AC's database, and this PR is still incorrect.

En'kilah Hatchling attacks the player, but the player can't attack and select, and I'm sure that's not true, so I modified it

cTrinity commented:
> > En'kilah Hatchling attacks the player, but the player can't attack and select, and I'm sure that's not true, so I modified it
> 
> I checked on Classic/AC, and yes, the hatchlings attack the player on AC but not on Classic, but the unit_flags are not the culprit. It may be the faction template, AC has 14 but the sniffed value is 634. I tested these creatures with the adjusted faction, and they did not attack. If you'll adjust your PR accordingly (including template 25389, as that is also an En'kilah Hatchling duplicate), it'd be good to go for me.

Okey, 

> > En'kilah Hatchling attacks the player, but the player can't attack and select, and I'm sure that's not true, so I modified it
> 
> I checked on Classic/AC, and yes, the hatchlings attack the player on AC but not on Classic, but the unit_flags are not the culprit. It may be the faction template, AC has 14 but the sniffed value is 634. I tested these creatures with the adjusted faction, and they did not attack. If you'll adjust your PR accordingly (including template 25389, as that is also an En'kilah Hatchling duplicate), it'd be good to go for me.

you mean by this change:
UPDATE `creature_template` SET `faction` = 634 WHERE `entry`  IN (25388, 25390);

heyitsbench commented:
> Okey, you mean by this change: UPDATE `creature_template` SET `faction` = 634 WHERE `entry` IN (25388, 25390);

Don't forget to include that additional hatchling duplicate template ID, so more like this:

```sql
UPDATE `creature_template` SET `faction` = 634 WHERE `entry` IN (25388, 25389, 25390);
```

Unless any more familiar contributors want to jump in and say there's a reason to exclude that one, I feel it should be included.

heyitsbench commented:
Not sure why there are duplicate templates for this particular creature, but that's a separate question for another time. Thanks for the PR. ❤️

heyitsbench commented:
Just noticed, you'll want to move the .sql file to `/data/sql/updates/pending_db_world` otherwise it won't get picked up.

cTrinity commented:
> Just noticed, you'll want to move the .sql file to `/data/sql/updates/pending_db_world` otherwise it won't get picked up.

I apologize for not noticing the path change.

cTrinity reviewed (commented):
fix(DB/Creature): Update En'kilah Hatchling faction

heyitsbench commented:
I'm not a maintainer, so my review doesn't really do anything for your PR, but given that I've tested it and it works and it lines up with sniffed data, I do feel that this is good to be merged.

This PR does also close #15241, if you add that into your PR description it'll automatically close the issue if/when your PR gets merged.

cTrinity commented:
> I'm not a maintainer, so my review doesn't really do anything for your PR, but given that I've tested it and it works and it lines up with sniffed data, I do feel that this is good to be merged.
> 
> This PR does also close #15241, if you add that into your PR description it'll automatically close the issue if/when your PR gets merged.

yes, thanks.

Gultask commented:
Thanks for the PR @cTrinity !
